* In ./src/core_atmosphere/physics/physics_sima:

davegill opened the pull request:
-> The "could be chemistry" fields to be mixed are made optional
  -> Files changed: bl_ysu.F bl_ysu.meta
  -> Fields changed: qmix (array of chem), qmixtnp (tendencies),
     nmix (number of chem constituents)

davegill committed:
* In ./src/core_atmosphere/physics/physics_sima:

  -> The "could be chemistry" fields to be mixed are made optional
  -> Files changed: bl_ysu.F bl_ysu.meta
  -> Fields changed: qmix (array of chem), qmixtnp (tendencies),
     nmix (number of chem constituents)

davegill commented:
@ldfowler58 
I tried to follow your convention for commits.

davegill committed:
* In ./src/core_atmosphere/physics/physics_sima:

  -> Make the surface ocean currents optional
  -> Files changed: bl_ysu.F bl_ysu.meta

davegill committed:
Use standard name for (Rv/Rd -1) from table

modified:   bl_ysu.meta

davegill committed:
Fortran syntax error: optional args can't use optional indexes

modified:   bl_ysu.F

davegill committed:
* In ./src/core_atmosphere/physics/physics_sima:

  -> Clean up indexing for outdented endif
  -> Files changed: bl_ysu.F

davegill committed:
* In physics_wrf and physics_sima

  -> Remove kts and kms from arg lists and metadata
  -> kts and kms are set as parameters: =1
  -> Change vertical index metadata to "dimension" from "end"
  -> Files changed: bl_ysu.F, bl_ysu.meta, module_bl_ysu.F

davegill committed:
* In physics/physics_sima and physics/physics_wrf:

  -> Remove the WRF-centric "j" index.
  -> Not used, available for debugging print out.
  -> Confusing to (i,k) or (k,i) models like CAM and MPAS.
  -> Files changed: bl_ysu.F, bl_ysu.meta, module_bl_ysu.F

davegill committed:
* In physics/physics_sima:

  -> conventionalize flag and option names
  -> "flag" is for T/F
  -> "option" is for an integer switch
  -> prefix is either "flag_" or "option_"
  -> units are defined as "flag"
  -> Files changed: bl_ysu.meta

ldfowler58 commented:
Hi Dave:

I am starting to look at all emails that you sent me, starting 18th June. I
understand that kts, kms are now set parameters and equal to 1 since we
assume that there is no need for optimization vertically. Would it also be
the case for running MPAS for the deep atmosphere, or in a WACCM mode, for
instance?

You could also argue that its should also be set as a parameter and removed
from the argument list as well. Do we know any model where the
decomposition is such that its is not equal to 1?

Thanks,
Laura


On Thu, Jun 18, 2020 at 8:15 AM Dave Gill <notifications@github.com> wrote:

> -> The "could be chemistry" fields to be mixed are made optional
> -> Files changed: bl_ysu.F bl_ysu.meta
> -> Fields changed: qmix (array of chem), qmixtnp (tendencies),
> nmix (number of chem constituents)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/ldfowler58/MPAS-Model/pull/11
> Commit Summary
>
>    - * In ./src/core_atmosphere/physics/physics_sima:
>
> File Changes
>
>    - *M* src/core_atmosphere/physics/physics_sima/bl_ysu.F
>    <https://github.com/ldfowler58/MPAS-Model/pull/11/files#diff-2eb519c8bbcf9e539652cb5a6b034065>
>    (11)
>    - *M* src/core_atmosphere/physics/physics_sima/bl_ysu.meta
>    <https://github.com/ldfowler58/MPAS-Model/pull/11/files#diff-a5c8d7879827e2060e8abc6b57518eca>
>    (6)
>
> Patch Links:
>
>    - https://github.com/ldfowler58/MPAS-Model/pull/11.patch
>    - https://github.com/ldfowler58/MPAS-Model/pull/11.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ldfowler58/MPAS-Model/pull/11>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA7LEXRKE3BOVRXLAX7QB2TRXIOPLANCNFSM4OBUDO2A>
> .
>


-- 
!-------------------------------------------------------------------------------------------------------------
Laura D. Fowler

Mesoscale and Microscale Meteorology Division (MMM)
National Center for Atmospheric Research
P.O. Box 3000, Boulder CO 80307-3000

e-mail: laura@ucar.edu
phone: 303-497-1628

!-------------------------------------------------------------------------------------------------------------

ldfowler58 commented:
Hi Dave:
For commit b992126, the integer nmix is not an optional index (at least in my sourcecode). So I do not understand the need to change the array declaration for qmix and qmixtnp. I did not recall having issues compiling the sourcecode.

davegill commented:
@ldfowler58 
Laura, 

> I understand that kts, kms are now set parameters and equal to 1 since we
assume that there is no need for optimization vertically. Would it also be
the case for running MPAS for the deep atmosphere, or in a WACCM mode, for
instance?

This must be OK since these are the recommendations from Steve Goldhaber. Additionally, I sent out a note to our local group asking about the removing "kts and kms", and no one responded.

>  the integer nmix is not an optional index (at least in my sourcecode). So I do not understand the need to change the array declaration for qmix and qmixtnp. I did not recall having issues compiling the sourcecode

The YSU scheme is a good example code. It is possible that the CAM guys may not have a variable similar to our qmix or qmixtnp. If we make them optional, then the CCPP will be able to generate the automated call.

Let me know if a "face-to-face" would be easier than this medium for discussion. I specifically made lots of little PRs to your branch for the purpose of you reviewing and questioning my ideas.

davegill committed:
* In physics/physics_sima:

  -> Files changed: bl_ysu.meta
  -> Swap "start" with "begin"
  -> remove "loopP in horiz dim name

davegill committed:
* In physics/physics_sima:

  -> Files changed: bl_ysu.meta
  -> Remove "_for" from gas_constant standard names
